spec/take_away_ui_spec.rb

```diff
@@ -0,0 +1,232 @@
+        expect(menu.basket).to eq dummy_basket
+      end
+
+      it 'puts dish selections into selection array' do
+        menu.select_dish("Tiramisu", 1)
+        expect(menu.basket.selection).to include selection2
+      end
+
+      it 'does not instantiate a new basket on second selections' do
+        menu.select_dish("Tiramisu", 1)
+        expect(menu.basket).to eq dummy_basket
+      end
+
+      it 'selecting dishes increases the size of selection array' do
+        expect {menu.select_dish("Tiramisu", 1)}.to change{menu.basket.selection.size}.by(1)
+      end
+
+
+
+      describe "#review_order" do
+        let(:sample_bill) {double :sample_bill}
+        let(:sample_total) {double :sample_total}
+
+        before do
+          allow(STDOUT).to receive(:puts).and_return("")
+          allow(dummy_basket).to receive(:itemised_bill).and_return sample_bill
+          allow(sample_bill).to receive(:each)
+          allow(dummy_basket).to receive(:total_bill).and_return sample_total
+        end
+
+        it 'will retrieve the itemised_bill' do
+          expect(dummy_basket).to receive(:itemised_bill).and_return sample_bill
+          menu.review_order
+        end
+
+        it 'will retrieve the total_bill' do
+          expect(dummy_basket).to receive(:total_bill).and_return sample_total
+          menu.review_order
+        end
+      end
+
+
+      describe "#confirm_order" do
+        let(:price) {double :price}
+        let(:sample_total) {double :sample_total}
+        let(:sample_selection) {[[:Carbonara, 3], [:Calzone, 1], [:Risotto, 3]]}
+
+
+        before do
+          allow(dummy_basket).to receive(:total_bill).and_return price
+          allow(dummy_basket).to receive(:checkout)
+          allow(dummy_basket).to receive(:itemised_bill).and_return sample_selection
+          allow(checkout_message).to receive(:send_sms)
+        end
+
+
+        it 'Stores the customer\'s expected price' do
+          menu.confirm_order(price)
+          expect(menu.price).to eq price
+        end
+
+        it 'checks the price against the total_bill' do
+          expect(sample_selection).to receive(:total_bill)
+          menu.confirm_order(price)
+        end
+
+        it 'raises error "Price does not match" if there is not a match.' do
+          expect{menu.confirm_order(sample_total)}.to raise_error("Price does not match")
+        end
+
+        it 'does not raise an error if prices match' do
+          allow(dummy_basket).to receive(:total_bill).and_return sample_total
+          expect{menu.confirm_order(sample_total)}.not_to raise_error
+        end
+
+        it 'checks-out the basket if prices match' do
+          expect(dummy_basket).to receive(:checkout)
+          menu.confirm_order(price)
+        end
+
+        it 'sends a confirmation text message' do
+          expect(takeaway).to receive(:send_sms).with(takeaway.order_summary)
+          takeaway.complete_order(takeaway.order_summary)
+        end
+
+
+    end
```

Comment:
Lint/BlockAlignment: `end` at 226, 4 is not aligned with `describe "#confirm_order" do` at 182, 6.